### ISSUE TITLE
How to make binary mask

### ISSUE THREAD

**Issue by hkaggarwal** [cagladbahadir/LOUPE#2]
Hello @cagladbahadir, @adalca, 

Thank you for making the version 2 of the code. It works very nicely.

During training, I am setting `sparsity=0.25` to get 25% undersampling in the `v2` code.

The training code (train_mri.py), successfully gives a probabilistic mask at the end: 
```
mask_model = keras.models.Model(model.inputs, model.get_layer('prob_mask_scaled').output)
prob_mask = mask_model.predict(xdata[0:1,...])

```
How can I get the **binary mask** out of `prob_mask` so that I can use it during testing? 
Should I threshold `prob_mask` by 0.5 or something? 
Currently, all the locations in the `prob_mask` are having non-zero values.

It will be great if you could help me with it.

Thank you.
Hemant

**Comment by adalca** [comment]
[forgive original answer, i misread the question]

You can sample from the probabilistic mask to get a real mask! :) 

so if in some location you have probability of 0.2, you can sample from a bernoulli with parameter p=0.2 . 

In practice, you can easily to this by sampling a `mask` from the uniform 0-1 distribution (e.g. with `np.random.random`), and then do `hard_mask = prob_mask < uniform_sample`

**Comment by hkaggarwal** [comment]
@adalca, Thank you so much, Adrian,  that is really helpful.

In case someone comes across this thread: 
In python, we can also sample from Bernoulli distribution as:
`hard_mask=np.random.binomial(1, prob_mask)`
since Binomial distribution with a single trial (n=1) is Bernoulli.

**Comment by adalca** [comment]
@hkaggarwal Cool. I guess we're old school and do it the manual way :D
